rs/ic_os/utils/src/lib.rs

```diff
@@ -14,70 +14,45 @@ pub fn get_command_stdout<'a, StringIter: IntoIterator<Item = &'a str>>(
     command: &str,
     args: StringIter,
 ) -> Result<String> {
-    let mut cmd = Command::new(command);
-    let mut arg_string: String = String::new();
-    // TODO - replace loop with single chain call pipeline.
-    for arg in args {
-        cmd.arg(arg);
-        arg_string.push_str(arg);
-        arg_string.push(' ');
-    }
-    let output = cmd.output()?;
+    let args_vec: Vec<&str> = args.into_iter().collect();
+
+    let output = Command::new(command).args(&args_vec).output()?;
     if !output.status.success() {
         bail!(
             "Error running command: '{} {}': {:?}",
             command,
-            arg_string,
+            args_vec.join(" "),
             output.stderr
         );
     }
     Ok(String::from_utf8(output.stdout)?)
 }
 
-/// Inject `to_inject` into a copy of `source` every `spacing` chars
-/// Will not inject at the end of the string.
-/// If spacing is 0, return source string
-/// TODO - use chunks for a more functional approach
-pub fn intersperse(source: &str, to_inject: char, spacing: usize) -> String {
-    if spacing == 0 {
-        return source.to_string();
-    }
-    let mut result = String::new();
-    for (i, c) in source.to_string().chars().enumerate() {
-        result.push(c);
-        if i % spacing == (spacing - 1) && i != (source.len() - 1) {
-            result.push(to_inject);
-        }
-    }
-    result
-}
-
-// Retry the given function `f` until either:
-// * f has been called `attempts` times
-// * `stop_pred` returns true when passed the result of `f()`
-// `wait_func` is called before each attempt.
-// The result returned is either the one held after `stop_pred` returns true or the one held after `attempts` has been breached.
+/// Retry the given function `f` until either:
+/// * f has been called `attempts` times
+/// * `stop_pred` returns true when passed the result of `f()`
+///
+/// The result returned is either the one held after `stop_pred` returns true or the one held after `attempts` has been breached.
 pub fn retry_pred<F, P, T, W>(attempts: usize, f: F, stop_pred: P, wait_func: W) -> Result<T>
 where
-    F: Fn() -> Result<T, anyhow::Error>,
+    F: Fn() -> Result<T>,
     P: Fn(&Result<T>) -> bool,
     W: Fn(usize),
 {
-    for attempt in 1..attempts - 1 {
-        // Final attempt is last line of function
```

Comment:
We should drop the final call to `f()`, below, since we have moved the wait.

Reply:
Discussed in person: The final call to f() is the last attempt so that it runs "attempt" times.

There's definitely a better way to do this, but this code should go away in the networking revamp anyway (it's only used in interfaces.rs), so I don't think we need to worry about it.